OHHTTPStubs/Sources/OHHTTPStubs.m

```diff
@@ -424,7 +424,24 @@ - (void)startLoading
                 // Notify if a redirection occurred
                 if (((responseStub.statusCode > 300) && (responseStub.statusCode < 400)) && redirectLocationURL)
                 {
-                    NSURLRequest* redirectRequest = [NSURLRequest requestWithURL:redirectLocationURL];
+                    NSMutableURLRequest* mReq = [self.request mutableCopy];
+                    switch (responseStub.statusCode)
+                    {
+                        case 301:
+                        case 302:
+                        case 307:
+                        case 308:
+                            //Preserve the original request method and set the new location URL
+                            [mReq setHTTPMethod:self.request.HTTPMethod];
+                            [mReq setURL:redirectLocationURL];
+                            break;
+
+                        default:
+                            break;
```

Comment:
So in summary the behaviour we want to implement I think is:

* 301, 302, 307, 307: do the `mutableCopy` + `mReq.URL = redirectLocationURL` + `copy` dance like you already do here
* default, especially for 303: `redirectRequest = [NSURLRequest requestWithURL:redirectLocationURL];` — instead of the copy (as, according to the RFC, 303 exists to tells that "the original request has been accepted and processed by the server and we now need to go to a new location to continue", so we must forget the old request (and not copy anything from it) and create a brand new one

Reply:
Got it. I'll make that change.  My code and your comment crossed on the wire.